packages/libs/error-stack/src/frame/mod.rs

```diff
@@ -177,77 +186,102 @@ impl Provider for Frame {
 
 impl fmt::Debug for Frame {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let field_name = match self.kind() {
-            FrameKind::Context => "context",
-            FrameKind::Attachment => "attachment",
-        };
-
-        fmt.debug_struct("Frame")
-            .field(field_name, &self.inner.vtable().unerase(&self.inner))
-            .field("location", &self.location)
-            .finish()
-    }
-}
-
-impl fmt::Display for Frame {
-    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt::Display::fmt(self.inner.vtable().unerase(&self.inner), fmt)
+        let mut debug = fmt.debug_struct("Frame");
+        match self.kind() {
+            FrameKind::Context(context) => {
+                debug.field("context", &context);
+            }
+            FrameKind::Attachment(Attachment::Printable(attachment)) => {
+                debug.field("attachment", &attachment);
+            }
+            FrameKind::Attachment(Attachment::Generic(_)) => (),
```

Comment:
I think we should print something here, especially on a debug impl you want to know there were attachments. Probably some standard placeholder like "GenericAttachment" or more explicitly say it doesn't impl Debug/Display